### ISSUE TITLE
'Waiting Device start' on iOS Simulators device

### ISSUE THREAD

**Issue by w243137321** [macacajs/app-inspector#185]
Environment check:

- app inspector version / 版本号:2.0.15

- os version / 系统:iOS 11.2

- device  information / 设备版本:Simulators iPhone X

- terminal log / 终端输出信息:

> try `app-inspector -u xxxx --verbose` for more log detail.
```
>> server.js:16:12 [master] pid:76147 server start with config:
 { port: 5678,
  verbose: true,
  udid: '41A07912-C4AB-45F0-8FDD-DD2E2A97FFAB',
  ip: '160.6.75.57',
  host: 'wmliMac.local',
  loaded_time: '2018-04-09 11:30:37' }
>> app-inspector.js:45:12 [master] pid:76147 server start at: http://160.6.75.57:5678
>> xctest-client.js:55:14 [master] pid:76147 project path: /usr/local/lib/node_modules/app-inspector/node_modules/_xctestwd@1.3.27@xctestwd/XCTestWD/XCTestWD.xcodeproj
>> xctest-client.js:247:14 [master] pid:76147 xcode version: 9.2
>> XCTestWD version: 1.3.27
>> Read simulator log at: /var/folders/kz/l363n36j4xb_stdkpv6s6yg00000gn/T/com.apple.dt.XCTest/IDETestRunSession-1B9B484E-ABEB-4778-8FBF-C342426AB963/XCTestWDUITests-5994FD02-528F-4C6D-A0E3-7D9D4BE2C7DF/StandardOutputAndStandardError.txt
>> Read simulator log at: /var/folders/kz/l363n36j4xb_stdkpv6s6yg00000gn/T/com.apple.dt.XCTest/IDETestRunSession-1B9B484E-ABEB-4778-8FBF-C342426AB963/XCTestWDUITests-5994FD02-528F-4C6D-A0E3-7D9D4BE2C7DF/StandardOutputAndStandardError.txt
>> xctest-client.js:195:18 [master] pid:76147 please check project: /usr/local/lib/node_modules/app-inspector/node_modules/_xctestwd@1.3.27@xctestwd/XCTestWD/XCTestWD.xcodeproj
>> xctest-client.js:195:18 [master] pid:76147 please check project: /usr/local/lib/node_modules/app-inspector/node_modules/_xctestwd@1.3.27@xctestwd/XCTestWD/XCTestWD.xcodeproj
>> xctest-client.js:195:18 [master] pid:76147 please check project: /usr/local/lib/node_modules/app-inspector/node_modules/_xctestwd@1.3.27@xctestwd/XCTestWD/XCTestWD.xcodeproj
>> xctest-client.js:281:14 [master] pid:76147 killing deviceLogProc pid: 76221
>> xctest-client.js:286:14 [master] pid:76147 killing runnerProc pid: 76151
>> xctest-client.js:206:16 [master] pid:76147 xctest client exit with code: 65, signal: null
>> xctest-client.js:119:18 [master] pid:76147 simulator log process exit with code: null, signal: SIGKILL
>> Read simulator log at: /var/folders/kz/l363n36j4xb_stdkpv6s6yg00000gn/T/com.apple.dt.XCTest/IDETestRunSession-1B9B484E-ABEB-4778-8FBF-C342426AB963/XCTestWDUITests-5994FD02-528F-4C6D-A0E3-7D9D4BE2C7DF/StandardOutputAndStandardError.txt
  <-- GET /
  --> GET / 200
  <-- GET /
  --> GET / 200
  <-- GET /
  --> GET / 200
  <-- GET /
  --> GET / 200
  <-- GET /
  --> GET / 200
  <-- GET /
  --> GET / 200
  <-- GET /
  --> GET / 200
```

**Comment by manish-algoworks** [comment]
Any update on this?

**Comment by SamuelZhaoY** [comment]
@manish-algoworks @w243137321  please refer to this https://github.com/macacajs/app-inspector/issues/184

**Comment by abhivaikar** [comment]
@SamuelZhaoY, @manish-algoworks 
I have the same issue. App inspector page just keeps showing "Waiting Device start" although iOS Simulator is running already. Issue #184 doesn't explain much on what needs to be done to resolve this issue. xctestwd on my system is 1.3.31. Does that mean I am supposed to downgrade it to 1.3.28?

This is the log that shows up when I start app-inspector:

```
>> app-inspector.js:45:12 [master] pid:4227 server start at: http://192.168.1.70:5678
>> xctest-client.js:55:14 [master] pid:4227 project path: /usr/local/lib/node_modules/app-inspector/node_modules/xctestwd/XCTestWD/XCTestWD.xcodeproj
>> xctest-client.js:247:14 [master] pid:4227 xcode version: 9.2
>> XCTestWD version: 1.3.31
>> Read simulator log at: /var/folders/q4/8s99_vb51l59q56rzpgctvv80000gn/T/com.apple.dt.XCTest/IDETestRunSession-F1331CAB-1670-417B-9F75-46FB996EF1F9/XCTestWDUITests-502D8618-040B-4AB5-81A7-0795D36FB114/StandardOutputAndStandardError.txt
>> xctest-client.js:195:18 [master] pid:4227 please check project: /usr/local/lib/node_modules/app-inspector/node_modules/xctestwd/XCTestWD/XCTestWD.xcodeproj
>> Read simulator log at: /var/folders/q4/8s99_vb51l59q56rzpgctvv80000gn/T/com.apple.dt.XCTest/IDETestRunSession-F1331CAB-1670-417B-9F75-46FB996EF1F9/XCTestWDUITests-502D8618-040B-4AB5-81A7-0795D36FB114/StandardOutputAndStandardError.txt
>> xctest-client.js:195:18 [master] pid:4227 please check project: /usr/local/lib/node_modules/app-inspector/node_modules/xctestwd/XCTestWD/XCTestWD.xcodeproj
>> xctest-client.js:195:18 [master] pid:4227 please check project: /usr/local/lib/node_modules/app-inspector/node_modules/xctestwd/XCTestWD/XCTestWD.xcodeproj
>> xctest-client.js:195:18 [master] pid:4227 please check project: /usr/local/lib/node_modules/app-inspector/node_modules/xctestwd/XCTestWD/XCTestWD.xcodeproj
>> xctest-client.js:281:14 [master] pid:4227 killing deviceLogProc pid: 4249
>> xctest-client.js:286:14 [master] pid:4227 killing runnerProc pid: 4230
>> xctest-client.js:206:16 [master] pid:4227 xctest client exit with code: 65, signal: null
>> xctest-client.js:119:18 [master] pid:4227 simulator log process exit with code: null, signal: SIGKILL
>> Read simulator log at: /var/folders/q4/8s99_vb51l59q56rzpgctvv80000gn/T/com.apple.dt.XCTest/IDETestRunSession-F1331CAB-1670-417B-9F75-46FB996EF1F9/XCTestWDUITests-502D8618-040B-4AB5-81A7-0795D36FB114/StandardOutputAndStandardError.txt
```

**Comment by abhivaikar** [comment]
UPDATE: I was able to get app-inspector up and running fine now.

1. Upgraded XCode to latest version.
2. Upgraded macOS.
3. Folllowed step 2 in #184 and then performed step 3.
4. I got a compilation error `Module compiled with Swift 4.0.3 cannot be imported in Swift 4.1.2` on XCode. To resolve this I followed this answer on SO: https://stackoverflow.com/a/47051670/1335594 and ran `carthage update --platform iOS` inside xctestwd folder as given in step 2 of #184.

Thanks @SamuelZhaoY.
